examples/tools/generate.ipynb

```diff
@@ -0,0 +1,502 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pycompwa.ui as pwa"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# specify kinematics\n",
+    "particle_list = pwa.read_particles('model.xml')\n",
+    "kinematics = pwa.create_helicity_kinematics('model.xml', particle_list)\n",
+    "\n",
+    "# speficy generators\n",
+    "generator = pwa.EvtGenGenerator(\n",
+    "    kinematics.get_particle_state_transition_kinematics_info())\n",
+    "random_generator = pwa.StdUniformRealGenerator(12345)\n",
+    "\n",
+    "# generate phase space sample\n",
+    "phsp_sample = pwa.generate_phsp(1000, generator, random_generator)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Intensity-based samples"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "Data samples are more complicated than phase space samples in that they\n",
+    "represent the intensity profile resulting from a decay. You therefore need an\n",
+    ":class:`.Intensity` object and a phase-space over which to generate that\n",
+    "intensity distribution. We therefore call such a data sample an\n",
+    "**intensity-based sample**.\n",
+    "\n",
+    "An intensity-based sample is generated with the function\n",
+    ":func:`~.ui.generate`. Its usage is similar to :func:`.generate_phsp`, but\n",
+    "now you have to give an :class:`.Intensity` and a :class:`.Kinematics`\n",
+    "instance as additional arguments. Sampling is performed through hit & miss,\n",
+    "which is reflected by the efficiency printed at the end."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create intensity profile\n",
+    "intensity_builder = pwa.IntensityBuilderXML(\n",
+    "    'model.xml', particle_list, kinematics, phsp_sample)\n",
+    "intensity = intensity_builder.create_intensity()\n",
+    "\n",
+    "# generate data\n",
+    "data_sample = pwa.generate(\n",
+    "    100, kinematics, generator, intensity, random_generator)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Importance sampling"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "In the example above, we used the decay $J/\\psi \\to \\gamma \\pi^0 \\pi^0$\n",
+    "with a f resonances only. Imagine, however, that we want to investigate the\n",
+    "resonance $\\omega \\to \\gamma\\pi^0$. This resonance has such a narrow peak\n",
+    "that we have to widen it to be visible and, even then, using an evenly\n",
+    "distributed phase space would not be feasible for a hit & miss strategy:\n",
+    "we want to have a higher density of phase-space points near the resonance.\n",
+    "\n",
```

Comment:
This is written a bit vaguely. The importance sampled phase space sample is generated via hit and miss. And it is a phase space sample which is needed for the normalization of the amplitude. So if you use a uniform phase space sample your approximation of the integral is either bad or you need a huge amount of phase space data to calculate it precisely. Therefore importance sampling